src/zsclient.cpp

```diff
@@ -597,7 +598,7 @@ namespace zsync2 {
             // URL might be relative -- we need an absolute URL to do a fetch
             std::string absoluteUrl;
 
-            if (!makeUrlAbsolute(referer, url, absoluteUrl)) {
+            if (!makeUrlAbsolute(referer, url, absoluteUrl) && url_alt.empty()) {
```

Comment:
Not a fan of this if/else construct. Basically, the empty check is repeated twice. It's possible to check only once by wrapping the existing check in another `if`. Please refactor.